tests/unit/test_operations/test_get_filtered_model_variables.py

```diff
@@ -0,0 +1,698 @@
+                            {"name": "--TERM", "ordinal": 1},
+                            {"name": "--SEQ", "ordinal": 2},
+                        ],
+                    },
+                    {
+                        "name": GENERAL_OBSERVATIONS_CLASS,
+                        "label": GENERAL_OBSERVATIONS_CLASS,
+                        "classVariables": [
+                            {
+                                "name": "STUDYID",
+                                "role": VariableRoles.IDENTIFIER.value,
+                                "ordinal": 1,
+                            },
+                            {
+                                "name": "DOMAIN",
+                                "role": VariableRoles.IDENTIFIER.value,
+                                "ordinal": 2,
+                            },
+                            {
+                                "name": "USUBJID",
+                                "role": VariableRoles.IDENTIFIER.value,
+                                "ordinal": 3,
+                            },
+                            {
+                                "name": "AETERM",
+                                "role": VariableRoles.IDENTIFIER.value,
+                                "ordinal": 4,
+                            },
+                            {
+                                "name": "VISITNUM",
+                                "role": VariableRoles.TIMING.value,
+                                "ordinal": 17,
+                            },
+                            {
+                                "name": "VISIT",
+                                "role": VariableRoles.TIMING.value,
+                                "ordinal": 18,
+                            },
+                            {
+                                "name": "TIMING_VAR",
+                                "role": VariableRoles.TIMING.value,
+                                "ordinal": 33,
+                            },
+                        ],
+                    },
+                ],
+            },
+            {
+                "_links": {"model": {"href": "/mdr/sdtm/1-5"}},
+                "classes": [
+                    {
+                        "name": "Events",
+                        "datasets": [
+                            {
+                                "name": "AE",
+                                "label": "Adverse Events",
+                                "datasetVariables": [
+                                    {"name": "AETEST", "ordinal": 1},
+                                    {"name": "AENEW", "ordinal": 2},
+                                    {"name": "VISITNUM", "ordinal": 3},
+                                    {"name": "VISIT", "ordinal": 4},
+                                ],
+                            }
+                        ],
+                    }
+                ],
+            },
+        )
+    ],
+)
+def test_get_filtered_model_variables1(
```

Comment:
The logic within these 4 test methods looks very similar. If this is the case, I think it would be better to parameterize the values that are different (key_name, key_value, variables, etc) and have 1 test method with multiple sets of arguments.
Example from the pytest documentation:
```python
@pytest.mark.parametrize("test_input,expected", [("3+5", 8), ("2+4", 6), ("6*9", 42)])
def test_eval(test_input, expected):
    assert eval(test_input) == expected
```

Reply:
I tried to do that and found out that it does not save any coding. I had to repeatedly define model_metadata and standard_metadata. I'd recommend to use the separate test cases.

Reply:
@htu I expect you would need to repeat the definitions for model_metadata and standard_metadata. That is what this example is showing. However, you wouldn't need to repeat the test method definitions and the code within the test methods.